tests/test_telemetry.py

```diff
@@ -369,3 +369,31 @@ def test_app_dependencies_loaded(self):
         for dependency, seen in seen_loaded_dependencies.items():
             if not seen:
                 raise Exception(dependency + " not recieved in app-dependencies-loaded message")
+
+    def setup_app_product_change(self):
+        weblog.get("/enable_product")
+
+    @missing_feature(
+        context.library in ("dotnet", "nodejs", "java", "python"),
+        reason="Weblog GET/enable_product and app-product-change event is not implemented yet.",
+    )
+    def test_app_product_change(self):
+        """Test product change data when product is enabled"""
+
+        def validator(data):
+            content = data["request"]["content"]
+            if content.get("request_type") == "app-product-change":
+                products = content["payload"]["products"]
+                for prod in products > 0:
+                    appsecEnabled = prod["appsec"]["enabled"]
+                    assert appsecEnabled == True, f"Product appsec enabled flag is {appsecEnabled}"
+                    profilerEnabled = prod["profiler"]["enabled"]
+                    assert profilerEnabled == True, f"Product appsec enabled flag is {profilerEnabled}"
+                    dynamicInstrumentationEnabled = prod["dynamic_instrumentation"]["enabled"]
+                    assert (
+                        dynamicInstrumentationEnabled == False
+                    ), f"Product appsec enabled flag is {dynamicInstrumentationEnabled}"
```

Comment:
better error message is like "dynamic_instrumentation was expected to be False, found True"